collect_app/src/androidTest/java/org/odk/collect/android/tasks/InstanceUploaderTaskTest.java

```diff
@@ -57,7 +57,7 @@ public void shouldUploadAnInstance() throws Exception {
         HEAD: {
             RecordedRequest r = nextRequestFor(server);
             assertEquals("HEAD", r.getMethod());
-            assertMatches("/submission\\?deviceID=imei%3A\\d+", r.getPath());
+            assertMatches("/submission\\?deviceID=\\w+%3A\\d+", r.getPath());
```

Comment:
Right, spoke too soon there. The escapes are just right. It's just that the identifier (part after the colon) can also include letters. 

`/submission\?deviceID=\w+%3A[a-zA-Z0-9]+` passes for me.